code/modules/overmap/planetoids/planetoid_skybox.dm

```diff
@@ -0,0 +1,112 @@
+/obj/effect/overmap/visitable/sector/planetoid/proc/get_skybox_secondary_surface()
+	if(!water_color)
+		return
+	var/image/secondary = image(skybox_icon, "water")
+	secondary.color = water_color
+	secondary.appearance_flags = PIXEL_SCALE
+	secondary.transform = secondary.transform.Turn(rand(0,360))
+	return secondary
+
+/// Generates the raw base/surface of the planetoid skybox image.
+/obj/effect/overmap/visitable/sector/planetoid/proc/generate_skybox_planetoid_surface()
+	var/image/surface   = image(skybox_icon, "")
+	var/image/primary   = get_skybox_primary_surface()
+	var/image/secondary = get_skybox_secondary_surface()
+	primary.layer = 1.0
+	surface.overlays += primary
+
+	if(secondary)
+		secondary.layer = 3.0
+		surface.overlays += secondary
+	return surface
+
+/// Generates the cloud cover over the planetoid.
+/obj/effect/overmap/visitable/sector/planetoid/proc/generate_skybox_planetoid_clouds()
+	if(!has_atmosphere())
+		return
+	var/image/clouds = overlay_image(skybox_icon, "weak_clouds", get_atmosphere_color() || COLOR_WHITE)
+	if(water_color) //#FIXME: This really should check if there's actually some liquid to make clouds from or something, instead of checking if there's a water overlay
+		clouds.overlays += image(skybox_icon, "clouds")
+	return clouds
+
+/// Generates the halo underlay around the planetoid if there's any.
+/obj/effect/overmap/visitable/sector/planetoid/proc/generate_skybox_planetoid_halo()
+	if(!has_atmosphere())
+		return
+	return image(skybox_icon, "atmoring")
+
+/// Generates the planetoid's shadow and rimlight overlays
+/obj/effect/overmap/visitable/sector/planetoid/proc/generate_skybox_planetoid_shading()
+	var/list/shading_overlays = list()
+	var/image/shadow = image(skybox_icon, "shadow")
+	shadow.blend_mode = BLEND_MULTIPLY
+	shading_overlays += shadow
+	shading_overlays += image(skybox_icon, "lightrim")
+	return shading_overlays
+
+/// Generate the planetoid's rings overlay if it has rings
+/obj/effect/overmap/visitable/sector/planetoid/proc/generate_skybox_planetoid_rings()
+	if(!has_rings())
+		return
+	var/image/rings = overlay_image('icons/skybox/planet_rings.dmi', get_ring_type_name(), get_ring_color())
+	//Need to offset the ring overlay because it's 512x512px, and the planet images are 256x256px
+	rings.pixel_x = -128
+	rings.pixel_y = -128
+	return rings
+
+/// Generates the skybox background displayed when a ship or viewer is in a sector above the planetoid
+/obj/effect/overmap/visitable/sector/planetoid/proc/generate_skybox_image()
+	//Everything is done in a separate proc for each layers, to allow for overriding for a variety of planetoids
+	skybox_image = generate_skybox_planetoid_surface()
+	ASSERT(skybox_image)
+	skybox_image.pixel_x = rand(0,64)
+	skybox_image.pixel_y = rand(128,256)
+	skybox_image.appearance_flags = RESET_COLOR
+
+	skybox_image.underlays += generate_skybox_planetoid_halo()
+	skybox_image.overlays  += generate_skybox_planetoid_clouds()
```

Comment:
The main idea was just to split those things in smaller more manageable bits. 
And the skybox is generated exactly once per exoplanets. So idk how much it would matter yeah. 
Is that something you'd want changed?

Reply:
I think I'll leave it like that for now.